src/server/api/routers/tdarr.ts

```diff
@@ -0,0 +1,375 @@
+import { TRPCError } from '@trpc/server';
+import axios from 'axios';
+import { z } from 'zod';
+import { checkIntegrationsType } from '~/tools/client/app-properties';
+import { getConfig } from '~/tools/config/getConfig';
+import { ConfigAppType } from '~/types/app';
+
+import { createTRPCRouter, publicProcedure } from '../trpc';
+
+const inputSchema = z.object({
```

Comment:
The `types/` directory seems to only contains TS types and interfaces, not Zod schemas. Those are kept next to the router definitions in the same file, only usually inline whereas mine are defined separately.

I'm not sure I get your second point. I'm parsing the response to be able to fail fast in case it doesn't conform to the schema.

Reply:
That's the point, using schema conformity like that you're failing it too fast even if all the data is correct, simply because we need to account for all the edge cases. This results in issues and support requests that are really hard to debug as they are too specific to reproduce.
I already caught one where I said you needed to add the "optional", but there are more to come for sure.
Since it's just data, it doesn't need to be individually verified.